extensions/amp-story/1.0/story-analytics.js

```diff
@@ -84,28 +86,49 @@ export class StoryAnalyticsService {
   onNavigationStateChange(stateChangeEvent) {
     switch (stateChangeEvent.type) {
       case StateChangeType.ACTIVE_PAGE:
-        this.triggerEvent(AnalyticsEvent.PAGE_VISIBLE);
+        this.triggerEvent(StoryAnalyticsEvent.PAGE_VISIBLE);
         break;
       case StateChangeType.BOOKEND_ENTER:
-        this.triggerEvent(AnalyticsEvent.BOOKEND_ENTER);
+        this.triggerEvent(StoryAnalyticsEvent.BOOKEND_ENTER);
         break;
       case StateChangeType.BOOKEND_EXIT:
-        this.triggerEvent(AnalyticsEvent.BOOKEND_EXIT);
+        this.triggerEvent(StoryAnalyticsEvent.BOOKEND_EXIT);
         break;
       case StateChangeType.LAST_PAGE:
-        this.triggerEvent(AnalyticsEvent.LAST_PAGE_VISIBLE);
+        this.triggerEvent(StoryAnalyticsEvent.LAST_PAGE_VISIBLE);
         break;
     }
   }
 
   /**
-   * @param {!AnalyticsEvent} eventType
+   * @param {!StoryAnalyticsEvent} eventType
    */
   triggerEvent(eventType) {
-    triggerAnalyticsEvent(
-      this.element_,
-      eventType,
-      this.variableService_.get()
-    );
+    this.element_.dispatchCustomEvent(eventType, this.getDetails_(eventType));
+  }
+
+  /**
+   * Consolidates count of event types per page and variables of the event.
+   * @param {!StoryAnalyticsEvent} eventType
+   * @private
+   * @return {!Object}
+   */
+  getDetails_(eventType) {
+    const vars = this.variableService_.get();
+    const pageId = vars['storyPageId'];
+
+    this.eventsPerPage_[pageId] = this.eventsPerPage_[pageId]
+      ? this.eventsPerPage_[pageId]
+      : {};
+
+    this.eventsPerPage_[pageId][eventType] = this.eventsPerPage_[pageId][
```

Comment:
This is a little hard to read. Maybe cache the `this.eventsPerPage_[pageId]` reference in a `const` and assign to that?

```js
const pageEvents = this.eventsPerPage_[pageId];
pageEvents[eventType] = pageEvents[eventType] ? pageEvents[eventType] : 0;
pageEvents[eventType]++;
```